### PR TITLE
Introduce metrics for unique usage of adviser integrations

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/metrics-exporter/issues/549

## This introduces a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

File: thoth/metrics_exporter/jobs/adviser.py [hunk 1]
```diff
@@ -64,3 +64,21 @@ def get_adviser_count_per_source_type(cls) -> None:
...
+
+    @classmethod
+    @register_metric_job
+    def get_uniquer_usage_count_per_source_type(cls) -> None:
+        """Get unique number of users per Thoth Integration provided."""
+        users_count_per_source_type = cls.graph().get_origin_count_per_source_type(distinct=True)
+        for thoth_integration in ThothAdviserIntegrationEnum._member_names_:
+
+            if thoth_integration in users_count_per_source_type:
+
```
Comment (fridex) [inline thoth/metrics_exporter/jobs/adviser.py:77]:
```suggestion
```

File: thoth/metrics_exporter/jobs/adviser.py [hunk 2]
```diff
@@ -64,3 +64,21 @@ def get_adviser_count_per_source_type(cls) -> None:
...
                 _LOGGER.debug("graphdb_adviser_count_per_source_type(%r)=%r", thoth_integration, 0)
+
+
+    @classmethod
+    @register_metric_job
+    def get_uniquer_usage_count_per_source_type(cls) -> None:
+        """Get unique number of users per Thoth Integration provided."""
+        users_count_per_source_type = cls.graph().get_origin_count_per_source_type(distinct=True)
+        for thoth_integration in ThothAdviserIntegrationEnum._member_names_:
+
```
Comment (fridex) [inline thoth/metrics_exporter/jobs/adviser.py:75]:
```suggestion
```

File: thoth/metrics_exporter/jobs/adviser.py [hunk 3]
```diff
@@ -64,3 +64,21 @@ def get_adviser_count_per_source_type(cls) -> None:
...
+            if thoth_integration in users_count_per_source_type:
+
+                counts = users_count_per_source_type[thoth_integration]
+                metrics.graphdb_users_count_per_source_type.labels(thoth_integration).set(counts)
+                _LOGGER.debug("graphdb_users_count_per_source_type(%r)=%r", thoth_integration, counts)
+            else:
+
```
Comment (fridex) [inline thoth/metrics_exporter/jobs/adviser.py:82]:
```suggestion
```

**Review by fridex — COMMENTED** [review]
Why was relocking needed?

LGTM otherwise.

**Comment by pacospace** [comment]
> Why was relocking needed?
> 
> LGTM otherwise.

I was due to that issue with pipenv with marker that requested `python version < 3.8` that we encountered also in another repository during this migration. If I remember well the issue was with `typing-extensions`

**Review by fridex — APPROVED** [review]
:+1: 

thanks! :100:

**Pull request merged**